misc.go

```diff
@@ -37,6 +37,45 @@ func (e *RateLimitedError) Error() string {
 	return fmt.Sprintf("Slack rate limit exceeded, retry after %s", e.RetryAfter)
 }
 
+func getFile(ctx context.Context, client HTTPRequester, url, token string, writer io.Writer) error {
+	body := &bytes.Buffer{}
+
+	req, err := http.NewRequest("GET", url, body)
+	req = req.WithContext(ctx)
+
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusTooManyRequests {
```

Comment:
maybe create a helper method for handling errors, this code is littered all over the place.